src/main/java/org/spongepowered/api/item/inventory/Inventory.java

```diff
@@ -0,0 +1,230 @@
+/*
+ * This file is part of Sponge, licensed under the MIT License (MIT).
+ *
+ * Copyright (c) SpongePowered.org <http://www.spongepowered.org>
+ * Copyright (c) contributors
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.spongepowered.api.item.inventory;
+
+import org.spongepowered.api.Nameable;
+import org.spongepowered.api.item.ItemType;
+import org.spongepowered.api.item.ItemTypes;
+import org.spongepowered.api.text.translation.Translatable;
+
+import com.google.common.base.Optional;
+
+/**
+ * Base interface for queryable inventories.
+ * 
+ * TODO Flesh out javadoc from proposal document. For now, see proposal doc
+ * here: https://github.com/SpongePowered/SpongeAPI/pull/443
+ */
+public interface Inventory extends Iterable<Inventory>, Nameable {
```

Comment:
I considered putting this on `Container` instead but in vanilla minecraft every `IInventory` is `IWorldNameable` and thus anything which is an Inventory basically has a name. I extracted that out to `Nameable` partly because it's a useful interface to have on its own merits but also because we could potentially graft the `Nameable` onto the hierarchy somewhere else if we don't like it being on the base interface.

Food for thought at any rate. Where I'm not sure about stufff I've erred on the side of simplicity and consistency with the view that we can throw it all in the bin if we feel the inclination.